services/src/unit-test/java/uk/gov/moj/sdt/services/TargetApplicationSubmissionServiceTest.java

```diff
@@ -1013,106 +951,114 @@ public void processCCDReferenceFailsOnUnSupportedRequest() {
-        EasyMock.expectLastCall();
-
         final IGlobalParameter connectionTimeOutParam = new GlobalParameter();
         connectionTimeOutParam.setName("TARGET_APP_TIMEOUT");
         connectionTimeOutParam.setValue("1000");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_TIMEOUT")).andReturn(
-            connectionTimeOutParam);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_TIMEOUT")).thenReturn(connectionTimeOutParam);
 
         final IGlobalParameter receiveTimeOutParam = new GlobalParameter();
         receiveTimeOutParam.setName("TARGET_APP_RESP_TIMEOUT");
         receiveTimeOutParam.setValue("12000");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT")).andReturn(
-            receiveTimeOutParam);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, "TARGET_APP_RESP_TIMEOUT")).thenReturn(receiveTimeOutParam);
 
         this.mockCmcConsumerGateway.individualRequest(individualRequest, 1000, 12000);
-        EasyMock.expectLastCall().andAnswer(() -> {
-            throw new CaseOffLineException("Case OffLine");
-        });
 
-        this.mockMessageWriter.queueMessage(EasyMock.isA(ISdtMessage.class), EasyMock.isA(String.class),
-                                            EasyMock.anyBoolean());
-        EasyMock.expectLastCall();
+        this.mockMessageWriter.queueMessage(isA(ISdtMessage.class), isA(String.class), anyBoolean());
 
         final IGlobalParameter maxForwardingAttemptsParam = new GlobalParameter();
         maxForwardingAttemptsParam.setName("MAX_FORWARDING_ATTEMPTS");
         maxForwardingAttemptsParam.setValue("3");
-        expect(this.mockCacheable.getValue(IGlobalParameter.class, "MAX_FORWARDING_ATTEMPTS")).andReturn(
-            maxForwardingAttemptsParam);
-
-        EasyMock.replay(mockIndividualRequestDao);
-        EasyMock.replay(mockCmcConsumerGateway);
-        EasyMock.replay(xmlReader);
-        EasyMock.replay(mockMessageWriter);
-        EasyMock.replay(ccdReferenceValidator);
-        EasyMock.replay(mockCacheable);
 
-        // Setup dummy target response
         SdtContext.getContext().setRawInXml("response");
 
         this.targetAppSubmissionService.processRequestToSubmit(sdtRequestRef, false);
 
-        EasyMock.verify(mockIndividualRequestDao);
-        EasyMock.verify(mockMessageWriter);
-        EasyMock.verify(mockCmcConsumerGateway);
-        EasyMock.verify(mockCacheable);
-        EasyMock.verify(xmlReader);
-        EasyMock.verify(ccdReferenceValidator);
-
         Assert.assertEquals("Individual Request status not as expected",
                             IIndividualRequest.IndividualRequestStatus.FORWARDED.getStatus(),
                             individualRequest.getRequestStatus());
+
+        verify(mockCmcConsumerGateway).individualRequest(any(IIndividualRequest.class), anyLong(), anyLong());
```

Comment:
Might need some additional verify() statements here depending on changes made to the rest of the test.

Reply:
Do we need to add a verification for this.mockIndividualRequestDao.getRequestBySdtReference() here as well? There's a when statement for it earlier in the test.

Reply:
added it now